src/components/genres/genres.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import {Link} from "react-router-dom";
+import {connect} from "react-redux";
+import PropTypes from "prop-types";
+
+const Genres = (props) => {
+  const {changeGenre, films, genre} = props;
+
+  let filmsGenres = [{
+    genre: `All genres`
+  }];
+
+  films.forEach(
+      (film) => {
+        film.genre.forEach((genreFilm) => {
+          if (!filmsGenres.some((filmsGenre) => {
+            return filmsGenre.genre === genreFilm;
+          })) {
+            filmsGenres.push({genre: genreFilm});
+          }
+        });
+      }
+  );
+
+  return (
+    <ul className="catalog__genres-list">
+      {filmsGenres.map((itemGenre, i) => (
+        <li
+          key={`genre-${i}`}
+          className={`catalog__genres-item ${itemGenre.genre === genre ? `catalog__genres-item--active` : ``}`}
+          onClick={(event) => {
+            event.preventDefault();
+            changeGenre(films, itemGenre);
+          }}
+        >
+          <Link to="#" className="catalog__genres-link">{itemGenre.genre}</Link>
```

Comment:
```suggestion
          <Link to="/" className="catalog__genres-link">{itemGenre.genre}</Link>
```
лучше # не использовать, Link не знает что это такое )